datasync/chaindatafetcher/kafka/kafka.go

```diff
@@ -164,7 +188,7 @@ func (k *Kafka) Publish(topic string, data interface{}) error {
 	}
 	segments, totalSegments := k.split(dataBytes)
 	for idx, segment := range segments {
-		msg := k.makeProducerMessage(topic, key, segment, uint64(idx), uint64(totalSegments))
+		msg := k.makeProducerMessage(topic, key, k.config.MsgVersion, k.config.ProducerId, segment, uint64(idx), uint64(totalSegments))
```

Comment:
`makeProducerMessage` doesn't need to pass `.kconfig.MsgVersion` and `k.config.ProducerId` because it  can access to the config also.

Reply:
You are right. I have updated. Thanks.